VCAC-A/script/setup_hddl.sh

```diff
@@ -1,110 +1,136 @@
-RUN mkdir -p /root/package
-RUN wget -P /root/package $py_link
-RUN wget -P /root/package $ov_link
-	
-#install python3.7
-RUN cd /root/package && \\
-    tar xzf Python-$py_ver.tgz && \\
-    cd Python-$py_ver && \\
-    ./configure && \\
-    make && make install
+FROM ubuntu:18.04 as builder
+ARG DOWNLOAD_LINK=https://github.com/libusb/libusb/archive/v1.0.22.tar.gz
+ARG INSTALL_DIR=/opt/intel/openvino
+ARG TEMP_DIR=/tmp/openvino_installer
+ADD $ov_link \$TEMP_DIR/openvino.tgz
+ADD \$DOWNLOAD_LINK \$TEMP_DIR/libusb.tar.gz
+RUN apt-get update && apt-get install -y --no-install-recommends \\
+    cpio sudo python3-pip python3-setuptools \\
+    libboost-filesystem1.65 libboost-thread1.65 libboost-program-options1.65 \\
+    libjson-c-dev build-essential autoconf automake libtool \\
+    kmod libelf-dev libusb-1.0-0 lsb-release && \\
+    rm -rf /var/lib/apt/lists/*
+RUN cd \$TEMP_DIR && \\
+    tar xf openvino.tgz && \\
+    cd l_openvino_toolkit* && \\
+    sed -i 's/decline/accept/g' silent.cfg && \\
+    ./install.sh -s silent.cfg
+RUN cd \$TEMP_DIR && \\
+    tar xvf libusb.tar.gz && \\
+    cd libusb-1.0.22/ && \\
+    ./autogen.sh enable_udev=no && \\
+    make -j \$(nproc) && \\
+    cp ./libusb/.libs/libusb-1.0.so /lib/x86_64-linux-gnu/libusb-1.0.so.0
+RUN cd /opt/intel/openvino/deployment_tools/tools/deployment_manager && \\
+    python3 deployment_manager.py --targets=hddl --output_dir=/root --archive_name=hddl
 
-#install openvino
-RUN cd /root/package && \\
-    tar xzf $openvino_name.tgz && \\
-    apt-get install -y cpio && \\
-    cd $openvino_name && \\
-    sed -i "s/ACCEPT_EULA=decline/ACCEPT_EULA=accept/" silent.cfg && \\
-    bash install.sh --ignore-signature --cli-mode -s silent.cfg
+FROM ubuntu:18.04
+RUN apt-get update && apt-get install -y sudo libboost-filesystem1.65 nasm \\
+    libboost-thread1.65 libboost-program-options1.65 libboost-all-dev libjson-c-dev libusb-1.0-0 \\
+    autoconf automake libtool kmod libelf-dev lsb-release && \\
+    apt-get clean && \\
+    rm -rf /var/lib/apt/lists/*
+COPY --from=builder /lib/x86_64-linux-gnu/libusb-1.0.so.0 /lib/x86_64-linux-gnu/libusb-1.0.so.0
+RUN mkdir -p /opt/intel
+COPY --from=builder /root/hddl.tar.gz /opt/intel
+RUN cd /opt/intel && tar xvf hddl.tar.gz && rm hddl.tar.gz
+COPY run_hddl.sh /usr/local/bin
+ENTRYPOINT ["/usr/local/bin/run_hddl.sh"]
+EOF
+}
 
-RUN cd /opt/intel/openvino/deployment_tools/tools/deployment_manager && \\
-    python3 deployment_manager.py --targets=hddl --output_dir=/root/package --archive_name=${package_name}
-RUN ls /opt/intel | grep  openvino_ > /root/package/ov_ver.log
+gen_entryfile()
+{
+	cat > "$1" <<EOF
+#!/bin/bash
+
+if [ "\$1" == "init" ]; then
+        modprobe i2c-dev i2c-i801 i2c-hid
+        cd /opt/intel/openvino/deployment_tools/inference_engine/external/hddl/drivers
+        bash setup.sh install
+	while true; do sleep 3600; done
+else
```

Comment:
is the else branch possible to run without privileges?

Reply:
Before we fix the USB pipe write and I2C ioexpander write permission issue, the "else" part still need privileged. That's next goal.